client/my-sites/plugins/plugins-list/index.tsx

```diff
@@ -230,20 +267,24 @@ export class PluginsList extends Component {
 		this.props.removePluginStatuses( 'completed', 'error', 'up-to-date' );
 	}
 
-	doActionOverSelected( actionName, action, selectedPlugins ) {
+	doActionOverSelected(
+		actionName: Action,
+		action: ( siteId: number, plugin: Plugin ) => void,
+		selectedPlugins?: Plugin[]
+	) {
 		if ( ! selectedPlugins ) {
 			selectedPlugins = this.props.plugins.filter( this.isSelected );
 		}
-		const isDeactivatingOrRemovingAndJetpackSelected = ( { slug } ) =>
+		const isDeactivatingOrRemovingAndJetpackSelected = ( slug: string ) =>
 			[ 'deactivating', 'activating', 'removing' ].includes( actionName ) && 'jetpack' === slug;
 
-		const flattenArrays = ( full, partial ) => [ ...full, ...partial ];
+		const flattenArrays = ( full: any[], partial: any[] ) => [ ...full, ...partial ];
 		this.removePluginStatuses();
 		const pluginAndSiteObjects = selectedPlugins
-			.filter( ( plugin ) => ! isDeactivatingOrRemovingAndJetpackSelected( plugin ) ) // ignore sites that are deactivating, activating or removing jetpack
-			.map( ( p ) => {
+			.filter( ( plugin: Plugin ) => ! isDeactivatingOrRemovingAndJetpackSelected( plugin.slug ) ) // ignore sites that are deactivating, activating or removing jetpack
+			.map( ( p: Plugin ) => {
 				return Object.keys( p.sites ).map( ( siteId ) => {
-					const site = this.props.allSites.find( ( s ) => s.ID === parseInt( siteId ) );
+					const site = this.props.allSites.find( ( s ) => s && s.ID === parseInt( siteId ) );
```

Comment:
We could as well use optional chaining:

```suggestion
					const site = this.props.allSites.find( ( s ) => s?.ID === parseInt( siteId ) );
```